docs/reference-guide/streams/streams.md

```diff
@@ -0,0 +1,93 @@
+---
+id: streams 
+title: Streams  
+sidebar_label: Streams 
+slug: /streams
+---
+
+Memgraph can connect to existing Kafka streams. To use streams, a user 
+must 
+- Manage a stream via a query
+- Provide a user defined transformation module
+
+More information about transformation modules can be found
+**[here](./transformation-modules/api/c-api.md)**
+
+The rest of this section describes how to manage streams with memgraph.
+
+## Creating a stream (query)
+The general syntax for creating a stream is:
+```cypher
+CREATE STREAM <stream name>
+  WITH TOPICS (['<topic1>', '<topic2>'])
+  WITH TRANSFORM <transform procedure>
+  [WITH CONSUMER_GROUP <consumer group name>]
+  [BATCH_INTERVAL <milliseconds>]
+  [BATCH_SIZE <size>];
+```
+Create a `STREAM` with name `<stream name>` that consumes messages from
+`TOPICS` with name `<topic1>` and `<topic2>`. `TRANSFORM` denotes the user 
+defined transformation with name `<transform procedure>`.
+
+Additionally, the user can provide the following optional parameters:
+- `CONSUMER_GROUP` with name `<consumer group name>`.
+- `BATCH_INTERVAL` denotes the maximum wait time interval for consuming message(s)
+before calling the transformation procedure with the already received message(s).
+This value must be greater than zero and is defaulted to 100.
+- `BATCH_SIZE` denotes the total number of messages to wait before calling 
+the transformation procedure with the already received message(s).  
+It must be greater than zero and is defaulted to 1000.
+
+The transformation procedure is called if either the `BATCH_INTERVAL` or the
+`BATCH_SIZE` is reached and there is at least one received message. 
+The `BATCH_INTERVAL` starts when the:
+- the stream is started
+- the processing of previous batch is completed
+- the previous batch interval ended without receiving any messages
+
+## Deleting a stream (query)
+```cypher
+DROP STREAM <stream name>;
+```
+Drops a stream with name `<stream name>` 
+
+## Start a stream (query)
+```cypher
+START STREAM <stream name> [LIMIT <count> BATCHES];
+START ALL STREAMS;
+```
+Starts a stream (or all streams) with name `<stream name>`.
+
+`<count>` denotes the total number of processed batches.
+
+When a stream is started, it should resume from the last committed 
+offset.
+
+## Stop a stream (query)
+```cypher
+STOP STREAM <stream name>;
+STOP ALL STREAMS;
+```
+Stops a stream (or all streams) with name `<stream name>`.
+
+## Show (query)
+```cypher
+SHOW STREAMS; 
+```
+Shows a list of existing streams with the following information:
+- stream name
+- list of topics
+- transformation procedure name
+- status
+- consumer group id
+- batch interval
+- batch size
+
+## Test connection (query)
+```cypher
+TEST STREAM <stream name> [LIMIT <count> BATCHES];
+```
+Tests the connection of `<stream name>` with `<count>` number of batches.
```

Comment:
Replace the word `test` with `check`. Same thing in the query and don't forget to add the new timeout config.